packages/compartment-mapper/src/parse-archive-cjs.js

```diff
@@ -0,0 +1,42 @@
+// @ts-check
+
+import { analyzeCommonJS } from '@endo/cjs-module-analyzer';
+import { encodeSyrup } from '@endo/syrup/encode';
+
+const textDecoder = new TextDecoder();
+
+const { freeze } = Object;
+
+/** @type {ParseFn} */
+export const parseCjs = async (
```

Comment:
I'm surprised that you're exporting under the exact same name here as in parse-cjs, when you distinguish `parsePreCjs`. Each of these two is just statically imported in one place, so you're not making any use of polymorphism between the two. Would `parseArchiveCjs` be clearer? (If it does not make things clearer, don't rename.)